File: internal/content/testdata/bench/go/nc-longest-consecutive.go
func longest_consecutive(nums []int) int {
	s := map[int]bool{}
	for _, n := range nums {
		s[n] = true
	}
	best := 0
	for n := range s {
		if !s[n-1] {
			length := 1
			for s[n+length] {
				length++
			}
			if length > best {
				best = length
			}
		}
	}
	return best
}

File: internal/content/testdata/bench/rust/nc-longest-consecutive.rs
use std::collections::HashSet;

fn longest_consecutive(nums: Vec<i64>) -> i64 {
    let s: HashSet<i64> = nums.into_iter().collect();
    let mut best = 0i64;
    for &n in &s {
        if !s.contains(&(n - 1)) {
            let mut length = 1i64;
            while s.contains(&(n + length)) {
                length += 1;
            }
            best = best.max(length);
        }
    }
    best
}
